### ISSUE TITLE
source shouldn't fail when no value found

### ISSUE THREAD

**Issue by olblak** [updatecli/updatecli#1963]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

For many plugin, if a source can't find a value then it triggers an error. 
For example:

<details><summary>updatecli.yaml</summary>

```
scms:
    github-action:
        kind: git 
        spec:
            password: '{{ requiredEnv "UPDATECLI_GITHUB_TOKEN" }}'
            url: https://github.com/actions/checkout.git
        disabled: false

sources:
    branch:
        name: Get latest branch for actions/checkout
        kind: gitbranch
        spec:
            versionfilter:
                kind: semver
                pattern: '*' 
        scmid: github-action
    release:
        name: Get latest GitHub Release for actions/checkout
        kind: githubrelease
        spec:
            owner: actions
            repository: checkout
            token: '{{ requiredEnv "UPDATECLI_GITHUB_TOKEN" }}'
            versionfilter:
                kind: semver
                pattern: '*' 
    tag:
        name: Get latest tag for actions/checkout
        kind: gittag
        spec:
            versionfilter:
                kind: semver
                pattern: '*' 
        scmid: github-action

```

</details>

<details><summary>Output</summary>

```
SCM repository retrieved: 1


++++++++++++++++++
+ AUTO DISCOVERY +
++++++++++++++++++



++++++++++++
+ PIPELINE +
++++++++++++



#############
# TEST.YAML #
#############


SOURCES
=======

release
-------
Searching for version matching pattern "*"
✔ GitHub release version "v4.1.1" found matching pattern "*" of kind "semver"


CHANGELOG:
----------

Release published on the 2023-10-17 15:53:17 +0000 UTC at the url https://github.com/actions/checkout/releases/tag/v4.1.1

## What's Changed
* Update CODEOWNERS to Launch team by @joshmgross in https://github.com/actions/checkout/pull/1510
* Correct link to GitHub Docs by @peterbe in https://github.com/actions/checkout/pull/1511
* Link to release page from what's new section by @cory-miller in https://github.com/actions/checkout/pull/1514

## New Contributors
* @joshmgross made their first contribution in https://github.com/actions/checkout/pull/1510
* @peterbe made their first contribution in https://github.com/actions/checkout/pull/1511

**Full Changelog**: https://github.com/actions/checkout/compare/v4.1.0...v4.1.1

branch
------
Searching for version matching pattern "*"
ERROR: no valid semantic version found
ERROR: ✗ filtering branches: no valid semantic version found
ERROR: {Get latest branch for actions/checkout ✗   {https://github.com/actions/checkout.git {main main main} }   Searching for version matching pattern "*"
ERROR: no valid semantic version found
ERROR: ✗ filtering branches: no valid semantic version found
} filtering branches: no valid semantic version found

tag
---
Searching for version matching pattern "*"
✔ Git tag "v4.1.1" found matching pattern "*" of kind "semver"

=============================

REPORTS:



- TEST.YAML:
	Source:
		✗ [branch] Get latest branch for actions/checkout
		✔ [release] Get latest GitHub Release for actions/checkout
		✔ [tag] Get latest tag for actions/checkout

Run Summary
===========
Pipeline(s) run:
  * Changed:	0
  * Failed:	0
  * Skipped:	1
  * Succeeded:	0
  * Total:	1
```

</details>

### Expected Behavior

I think similar comment to @dduportal [here](https://github.com/updatecli/updatecli/issues/1728#issuecomment-1782581812)

An error should only triggered be when something went wrong and we should accept empty values 

### Environment

```markdown
- OS:linux
- updatecli: 0.73.0
```

**Comment by dduportal** [comment]
At first glance, i disagree with this.

the rationale is that a source is at the core of a given updatecli flow and as such should fail the pipeline if it can’t find anything.

unless, of course, the end user opt out and accept empty response willingly as it would be an edge case.

I have a gut feeling  that we should check thoroughly the source plugins to differentiate errors (such as unauthorised github access) from source failing because empty (filter too strict, no remote candidate, etc)

**Comment by gionn** [comment]
what about adding a `continue-on-empty` property to a source which doesn't make updatecli exit with failure but should also prevent any change on targets where such source is referenced?

I agree that for hard failures (like auth) an error should triggered regardless.

**Comment by olblak** [comment]
Thank you both for your feedback. 
Let me explain how I ended up here. 

While adding Flux support, I realize that I couldn't handle Git repository as it has a similar update pipeline than GitHub Action I realized that I was missing the conditional target (https://github.com/updatecli/updatecli/issues/1385), so I decided to implement it so I could have the following pipeline.

![diagram](https://github.com/updatecli/updatecli/assets/2360224/a293f945-582c-4894-ae91-30e949020172)

Obviously we only expect one condition to succeed that define what type of value need to be used for our target. unfortunately two sources fails from time to time if no value could be retrieved based on the version filter rule :( 
Hence I was wondering if it makes sense to fail on empty value, which @dduportal confirm it's needed

Maybe I could add a flag to ignore source failure or maybe the truth is out there
